cmake/libutils.cmake

```diff
@@ -196,18 +197,29 @@ MACRO(MERGE_STATIC_LIBS TARGET OUTPUT_NAME LIBS_TO_MERGE)
       )  
     ELSE()
       # Generic Unix, Cygwin or MinGW. In post-build step, call
-      # script, that extracts objects from archives with "ar x" 
-      # and repacks them with "ar r"
+      # script, that uses a MRI script to append static archives.
+      SET(MRI_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.mri")
+      SET(MRI_SCRIPT_TPL "${MRI_SCRIPT}.tpl")
+
+      SET(SCRIPT_CONTENTS "CREATE $<TARGET_FILE:${TARGET}>\n")
+      FOREACH(LIB ${STATIC_LIBS})
+        STRING(APPEND SCRIPT_CONTENTS "ADDLIB ${LIB}\n")
+      ENDFOREACH()
+      STRING(APPEND SCRIPT_CONTENTS "SAVE\nEND\n")
+      FILE(WRITE ${MRI_SCRIPT_TPL} "${SCRIPT_CONTENTS}")
+      FILE(GENERATE OUTPUT ${MRI_SCRIPT} INPUT ${MRI_SCRIPT_TPL})
+
       ADD_CUSTOM_COMMAND(TARGET ${TARGET} POST_BUILD
+        DEPENDS ${MRI_SCRIPT}
         COMMAND ${CMAKE_COMMAND}
-          -DTARGET_LOCATION="$<TARGET_FILE:${TARGET}>"
-          -DTARGET="${TARGET}"
-          -DSTATIC_LIBS="${STATIC_LIBS}"
-          -DCMAKE_CURRENT_BINARY_DIR="${CMAKE_CURRENT_BINARY_DIR}"
+        ARGS
+          -DTARGET_SCRIPT="${MRI_SCRIPT}"
           -DCMAKE_AR="${CMAKE_AR}"
-          -DCMAKE_RANLIB="${CMAKE_RANLIB}"
           -P "${MYSQL_CMAKE_SCRIPT_DIR}/merge_archives_unix.cmake"
```

Comment:
Do you really need to call  cmake -P  here . That script after your patch contains a single EXECUTE_PROCESS.   Maybe instead  the script, you can do whatever you want to do, right here, inside ADD_CUSTOM_COMMAND, and remove merge_archives_unix.cmake ?

Reply:
would of been nice. `add_custom_command`  (https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) didn't have an INPUT_FILE specifier and couldn't see how to do a scriptname on the `ar` command line.